python/cudf/cudf/_lib/json.pyx

```diff
@@ -106,14 +106,20 @@ cpdef read_json(object filepaths_or_buffers,
         opts.set_dtypes(c_dtypes_map)
 
     # Read JSON
-    cdef cudf_io_types.table_with_metadata c_out_table
+    cdef cudf_io_types.table_with_metadata c_result
 
     with nogil:
-        c_out_table = move(libcudf_read_json(opts))
+        c_result = move(libcudf_read_json(opts))
 
-    column_names = [x.decode() for x in c_out_table.metadata.column_names]
-    return data_from_unique_ptr(move(c_out_table.tbl),
-                                column_names=column_names)
+    meta_names = [name.decode() for name in c_result.metadata.column_names]
+    df = cudf.DataFrame._from_data(*data_from_unique_ptr(
```

Comment:
I guess you moved the `_from_data` call into the Cython because `update_struct_field_names` has to happen in Cython? In the long term I think that probably indicates some level of restructuring is required, but we can deal with that when we get around to cuIO/Cython refactoring more broadly.